### ISSUE TITLE
Segmentation fault when build and execute environment don't match

### ISSUE THREAD

**Issue by jxiao1** [intel/linux-sgx#508]
If we build the enclave.so in the build server with SDK2.5, and sign it in another server which has SDK2.7, then maybe run the final application in the server with SDK2.5, which is not the same as signing environment,  there will be "Segmentation fault" when calling sgx_create_enclave().

This may be a known issue, but I still prefer to report it to find out whether there are some workarounds to avoid this kind of issue, for example, Could enclave.so include SDK version? if so, the sgx_sign tool which also with version information can compare the version in so file.

**Comment by lzha101** [comment]
Yes, this is a known issue. Building and signing should be with a same SDK to make sure everything OK.  And we will see how to report this mix-usage case in sgx_sign.
